src/main/java/uk/gov/hmcts/reform/preapi/services/CaseService.java

```diff
@@ -266,12 +271,19 @@ public void closePendingCases() {
     @Transactional(propagation = Propagation.REQUIRED, rollbackFor = Exception.class)
     public void onCaseClosed(Case c) {
         log.info("onCaseClosed: Case({})", c.getId());
-        var notifications = shareBookingService.deleteCascade(c)
+        var shares = shareBookingService.deleteCascade(c);
+        var notifications = shares
             .stream()
             .map(share -> new CaseStateChangeNotificationDTO(EmailType.CLOSED, c, share))
             .toList();
```

Comment:
This could be moved to inside the `if (!emailServiceBroker.isEnabled())` block since it only needs to run when using the old notification system. This happens a few times in this class- they can all be changed.